source/Files.h

```diff
@@ -33,41 +33,42 @@ class Files {
 	static void Init(const char * const *argv);
 
 	// The game's installation directory, or whichever directory was passed on the command line via `--resources`
-	static const std::string &Resources();
+	static const std::filesystem::path &Resources();
 	// The user-specific configuration directory, or whichever directory was passed on the command line via `--config`
-	static const std::string &Config();
+	static const std::filesystem::path &Config();
 
-	static const std::string &Data();
-	static const std::string &Images();
-	static const std::string &Sounds();
-	static const std::string &Saves();
-	static const std::string &Tests();
+	static const std::filesystem::path &Data();
+	static const std::filesystem::path &Images();
+	static const std::filesystem::path &Sounds();
+	static const std::filesystem::path &Saves();
+	static const std::filesystem::path &UserPlugins();
+	static const std::filesystem::path &GlobalPlugins();
+	static const std::filesystem::path &Tests();
 
 	// Get a list of all regular files in the given directory.
-	static std::vector<std::string> List(std::string directory);
+	static std::vector<std::filesystem::path> List(const std::filesystem::path &directory);
 	// Get a list of any directories in the given directory.
-	static std::vector<std::string> ListDirectories(std::string directory);
+	static std::vector<std::filesystem::path> ListDirectories(const std::filesystem::path &directory);
 	// Get a list of all regular files in the given directory or any directory
 	// that it contains, recursively.
-	static std::vector<std::string> RecursiveList(const std::string &directory);
-	static void RecursiveList(std::string directory, std::vector<std::string> *list);
+	static std::vector<std::filesystem::path> RecursiveList(const std::filesystem::path &directory);
 
-	static bool Exists(const std::string &filePath);
-	static std::time_t Timestamp(const std::string &filePath);
-	static void Copy(const std::string &from, const std::string &to);
-	static void Move(const std::string &from, const std::string &to);
-	static void Delete(const std::string &filePath);
+	static bool Exists(const std::filesystem::path &filePath);
+	static time_t Timestamp(const std::filesystem::path &filePath);
```

Comment:
```suggestion
	static std::time_t Timestamp(const std::filesystem::path &filePath);
```

Reply:
What's the difference between time_t and std::time_t?